spatialprofilingtoolbox/db/exchange_data_formats/metrics.py

```diff
@@ -76,17 +71,27 @@ class PhenotypeCounts(BaseModel):
 
 
 class ProximityMetricsComputationResult(BaseModel):
-    """
-    The response to a request for retrieval of proximity metrics in some specific case. This request
+    """The response to a request for retrieval of proximity metrics in some specific case. This request
     may also be a request for computation of these metrics in the background (which may be pending).
     """
     values: dict[str, float | None]
     is_pending: bool
 
 
+class SquidpyMetricsComputationResult(BaseModel):
+    """Response to an on demand request for computation of Squidpy metrics."""
+
+    values: dict[
+        str,
+        dict[str, list[float] | list[int]] |
+        dict[str, list[float]] |
+        dict[str, list[list[float]] | list[float] | list[int]]
+    ]
+    is_pending: bool
+
```

Comment:
Just a comment: These type definitions in `exchanged_data_formats` are designed specifically for the final API. The priority here is making the API docs readable/understandable by casual browsers, and to make it easier to read/review the API handlers in `app/main.py`.
The defs are in many cases extensively decoupled from the `scstudies` schema, where the priorities are different.
The point of my comment is that more-internal functions (like the providers' logic) may be better suited to types that are not the `exchanged_data_formats` types. This is a case-by-case basis thing.